cg/store/api/base.py

```diff
@@ -1,27 +1,134 @@
 """All models aggregated in a base class"""
-from attr import dataclass
+from typing import Any, Type
 
-from cg.store import models
+from alchy import Query, ModelBase
+from dataclasses import dataclass
+
+from cg.store.models import (
+    Analysis,
+    Application,
+    ApplicationVersion,
+    Bed,
+    BedVersion,
+    Collaboration,
+    Customer,
+    Delivery,
+    Family,
+    FamilySample,
+    Flowcell,
+    Invoice,
+    Organism,
+    Panel,
+    Pool,
+    Sample,
+    User,
+)
 
 
 @dataclass
 class BaseHandler:
     """All models in one base class"""
 
-    User = models.User
-    Bed = models.Bed
-    BedVersion = models.BedVersion
-    Customer = models.Customer
-    Collaboration = models.Collaboration
-    Sample = models.Sample
-    Family = models.Family
-    FamilySample = models.FamilySample
-    Flowcell = models.Flowcell
-    Analysis = models.Analysis
-    Application = models.Application
-    ApplicationVersion = models.ApplicationVersion
-    Panel = models.Panel
-    Pool = models.Pool
-    Delivery = models.Delivery
-    Invoice = models.Invoice
-    Organism = models.Organism
+    Analysis: Type[ModelBase] = Analysis
+    Application: Type[ModelBase] = Application
+    ApplicationVersion: Type[ModelBase] = ApplicationVersion
+    Bed: Type[ModelBase] = Bed
+    BedVersion: Type[ModelBase] = BedVersion
+    Collaboration: Type[ModelBase] = Collaboration
+    Customer: Type[ModelBase] = Customer
+    Delivery: Type[ModelBase] = Delivery
+    Family: Type[ModelBase] = Family
+    FamilySample: Type[ModelBase] = FamilySample
+    Flowcell: Type[ModelBase] = Flowcell
+    Invoice: Type[ModelBase] = Invoice
+    Organism: Type[ModelBase] = Organism
+    Panel: Type[ModelBase] = Panel
+    Pool: Type[ModelBase] = Pool
+    Sample: Type[ModelBase] = Sample
+    User: Type[ModelBase] = User
+
+    @staticmethod
+    def _get_query(table: Any) -> Query:
+        return table.query
+
+    def samples_to_receive(self, external=False) -> Query:
+        """Fetch incoming samples."""
+        return (
+            self._get_query(table=Sample)
+            .join(
+                Sample.application_version,
+                ApplicationVersion.application,
+            )
```

Comment:
This is part of the refactoring I am doing. It is basically a filter and a join query. I think the filtering and get should be done in status.py, but it is maybe good to have the join query here.

and then add the join query to the function in status.py

```suggestion
            def join_sample_ApplicationVersion(self) -> Query:
                return self._get_query(table=Sample).join(Sample.application_version, ApplicationVersion.application,)
```

Reply:
Good input 👍